cmd/gate/root.go

```diff
@@ -65,18 +66,20 @@ func initConfig() {
 	v.AutomaticEnv() // read in environment variables that match
 	v.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
 
-	if cfgFile := v.GetString("config"); cfgFile != "" {
+	cfgFile := v.GetString("config")
+	if cfgFile != "" {
 		v.SetConfigFile(cfgFile)
 	} else {
 		v.SetConfigName("config")
 		v.AddConfigPath(".")
 	}
 
-	// If a config file is found, read it in.
-	err := v.ReadInConfig()
-	if err != nil {
-		fmt.Println(fmt.Errorf("Error reading config file: %s \n", err))
-	} else {
+	switch err := v.ReadInConfig(); err.(type) {
+	case nil:
 		fmt.Println("Using config file:", v.ConfigFileUsed())
+	case viper.ConfigFileNotFoundError: // Do Nothing (config is optional)
+	default:
+		fmt.Println(fmt.Errorf("Error reading config file: %s \n", err))
```

Comment:
The problem would be that Gate would start nevertheless, ignoring this error, which is not good if one has explicitly set the flag.